packages/edit-post/src/components/layout/index.js

```diff
@@ -136,6 +129,26 @@ function Layout() {
 		}
 	}, [ isInserterOpen, isHugeViewport ] );
 
+	// Local state for save panel.
+	const [ isEntitiesSavedStatesOpen, setEntitiesSavedStates ] = useState(
+		false
+	);
+	const [
+		closeEntitiesCallback,
+		setCloseEntitiesCallback,
+	] = useState( () => () => {} );
```

Comment:
Why not just hold the callback? It's truthy.

Reply:
It's an odd quirk of `useState`.  If we gave it `() => {}` it would accept `{}` as the initial value and on use we would get `closeEntitiesCallback is not a function`.    [Here are a couple explanations](https://stackoverflow.com/questions/55621212/is-it-possible-to-react-usestate-in-react) from the overflow.

Reply:
I meant, why don't we have a single state value?

```js
const [ entitiesSavedStatesCallback, setEntitiesSavedStatesCallback ] = useState();
```

`entitiesSavedStatesCallback` would be falsy when the panel should be closed, and hold the truthy function when it is open.

Reply:
The callback only gets set when used from the publish/update button.  When we go through the a11y button flow to open the save panel we don't need / don't set any callback.  So instead of checking to see if the callback is truthy before firing it, we initialize it as something that can handle being fired in the first place.

Do you think it would make more sense to initialize this as falsy and change the close function similarly:
```
	const closeEntitiesSavedStates = useCallback(
		( callbackArg ) => {
			closeEntitiesCallback && closeEntitiesCallback( callbackArg );
			setIsEntitiesSavedStatesOpen( false );
		},
		[ closeEntitiesCallback ]
	);
```
instead?

Reply:
`isEntitiesSavedStatesOpen` doesn't need to exist.

To open the panel without a callback, just set it to a no-op or `true` and then do a `typeof` check before calling it in `closeEntitiesSavedStates`.

Reply:
That makes sense.  I found an interesting edge case I had to fix while implementing this (although not related to doing it this way).  The fix is in 76e1f70 .

1.  I never reset the callback when closing the panel, this would mean that we could save through the 'publish' button flow, make edits, then use the save a11y flow that would try to use the same callback still and sometimes error.

2.  It seems that the `useCallback` for `dismissPanel` was causing the callback from the publish button to be skipped.  There would be no error, but the set callback would not fire and it would move on to just closing the panel instead.  This means the reset noted in 1 above would not happen when dismissing the panel from the close (X) button.